File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    var answer = [];
    let num=numbers.split('');
    // 가능한 모든 숫자의 순열 찾기
    const getPermutations=(fixed,arr)=>{
        // arr: 아직 선택 안된 수 배열
        // fixed: 선택된수
        if(arr.length>=1){
            for(let i=0;i<arr.length;i++){
                const newNum = fixed+ arr[i];
                const copyArr=[...arr];
                copyArr.splice(i,1);
                if(!answer.includes(+newNum) && isPrimeNum(+newNum)){
                    answer.push(+newNum);
                }
                getPermutations(newNum,copyArr);
            }
        }
    }
    
    // 찾은 순열에 대해, 소수인지 확인
    
    const isPrimeNum=(n)=>{
        if(n<=1) return false;
        for(let i=2;i*i<=n;i++){
            if(n%i===0) return false;
        }
        return true;
    }
    getPermutations('',num);
    return answer.length;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <math.h>
#include <iostream>

using namespace std;

bool isPrime(int n){
    if(n<=1) return false;
    for(int i=2;i<=sqrt(n);i++){
        if(n%i==0) return false;
    }
    return true;
}
bool checked[9999999+1];
int solution(string numbers) {
    int answer = 0;
    sort(numbers.begin(),numbers.end());
    
    do{
        string str="";
        for(int i=0;i<numbers.size();i++){
            str+=numbers[i];
            int num=stoi(str);
            if(!checked[num]&&isPrime(num)){
                answer++;
                checked[num]=true;
            }
        }
    }while(next_permutation(numbers.begin(),numbers.end()));
    return answer;
}